lib/algsc.gi

```diff
@@ -780,15 +755,11 @@ InstallGlobalFunction( QuaternionAlgebra, function( arg )
               filter );
       SetFilterObj( A, IsAlgebraWithOne );
 #T better introduce AlgebraWithOneByStructureConstants?
+      return A;
+    end );
 
-      # Store the data for the next call.
-      if IsHPCGAP then
-        A := StoreQuaternionAlgebraData( a, b, F, A );
-      else
-        Add( QuaternionAlgebraData, [ a, b, FamilyObj( F ), A ] );
-      fi;
-
-    fi;
+    A:= AlgebraWithOne( F, GeneratorsOfAlgebra( stored ), "basis" );
```

Comment:
Here a new algebra gets created; setting the filter `IsFullSCAlgebra` would be useful.

Reply:
Great catch, fixed now, and commit message adjusted. I agree this is a bugfix.